compiler/ic/dce.nim

```diff
@@ -9,7 +9,7 @@
 
 ## Dead code elimination (=DCE) for IC.
 
-import std / [intsets, tables]
+import intsets, tables
```

Comment:
why? compiler/ic/dce.nim and intsets are not in the same nimble package

Reply:
Maybe we need csources_v1_2 instead but then I need to go through all the testing trouble *again*.

Reply:
> Bootstrapping on the BSDs doesn't work otherwise.

is a `--lib:lib` missing from compilation command in bootstrap? this should be fixable, at least i don't see why it couldn't work; i can take a look if needed; 

> Maybe we need csources_v1_2 instead

I'm still strongly arguing in favor of latest stable release (1.4.4), (see sample reasons in https://github.com/timotheecour/Nim/issues/251) and I'm ready to help if that's what it takes to make it happen. But 1_2 is still better than 1_0 at least.

This PR has tons of unrelated changes to the csources_v1 topic, can't it be 2 PRs?

Reply:
Every change is related to bootstrapping on the BSDs. I presume there is a regression for the 1.0 line where it's pickier about 'std' than even 0.9.x is.

Reply:
Also, Nim v1.0 can compile 1.2, 1.4, etc, whereas Nim 1.4 cannot even compile Nim 1.2, last time I checked.

Reply:
> Also, Nim v1.0 can compile 1.2, 1.4, etc

at the price of contorsions that prevent code simplifications or using newer features that aid in development of new features/bug fixes; many times the obvious way to write things didn't work because of boostrap. All that's needed is that each version of nim can be bootstrapped from a deterministic csources version (determined by a stored hash, see https://github.com/nim-lang/Nim/pull/16282#issuecomment-823445782)

Reply:
> Every change is related to bootstrapping on the BSDs. I presume there is a regression for the 1.0 line where it's pickier about 'std' than even 0.9.x is.

see https://github.com/nim-lang/Nim/pull/17801, how do i make bsd CI run in that PR?

Reply:
indeed, as I suspected, this is fixable simply by passing `--lib:lib` during bootstrap, see 
https://github.com/nim-lang/Nim/pull/17902 for the fix + explanation